src/shared.rs

```diff
@@ -153,15 +157,15 @@ impl From<&WorkspaceType> for String {
 //     }
 // }
 
-impl From<i32> for WorkspaceType {
-    fn from(int: i32) -> Self {
+impl TryFrom<i32> for WorkspaceType {
```

Comment:
Isn't better to leave `From<i32>` and add `TryFrom<i32>`? I mean someone wants to use simple `from(i32)` or `into()` instead of `try_from(i32)` and `try_into()` and vice versa.

Because of this code is for user, I think that better provide two variants which can be useful in different situations.

Reply:
I guess so -- I just didn't want it to panic

Reply:
It won't let me impl them both though

Reply:
Oh, I understand. Let me think about it

Reply:
I would like 0 panics in the codebase

Reply:
Okay, I leave it